### DISCUSSION TITLE
Open a workload item from within the workload/application

### DISCUSSION THREAD

**Discussion by dhrumil83** [microsoft/Microsoft-Fabric-workload-development-sample#108] [Q&A] [answered]
Wanted to check if any options were available to handle the use case described below.

In the current workflow, I can list all the lakehouses and handle the case when user clicks on one: 
`callDatahubOpen => openDialog` 
https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/src/controller/SampleWorkloadController.ts#L291-L322

Rather than listing all lakehouses and selecting one, I want to list all workload items we created(e.g. `Org.SampleWorkload.SampleWorkloadItem`), and when the user clicks on one, use `workloadClient.navigation.navigate` to navigate to that item.

Is something available to list all `Org.SampleWorkload.SampleWorkloadItem`?

Note: I tried changing supportedTypes in that function from `Lakehouse` to `Org.SampleWorkload.SampleWorkloadItem` but it did not work.

**Comment by ramizsha** [comment, marked as answer]
Regarding the navigation between items - this is not supported yet, we might support it in the future but currently there's no ETA.
Regarding listing the items, we are working on a Fabric public API to list all the items, ETA is end of November.

**Reply by dhrumil83** [reply]
> Regarding listing the items, we are working on a Fabric public API to list all the items, ETA is end of November.

@ramizsha Wondering if this was released, and if there is any doc on how to use it?

**Reply by anitamayorov** [reply]
Hi @dhrumil83, currently supported are admin public APIs - list and get item. For user APIs, list will be supported mid. January.
